src/test/link-rewriter.test.js

```diff
@@ -82,13 +82,74 @@ test.serial('rewriteLinks updates topic link to module', async t => {
 		assertTopic: (instanceUrl, orgUnit, { module, topic, data }) => {
 			t.is(instanceUrl, 'https://example.com/');
 			t.deepEqual(orgUnit, OrgUnit);
-			t.deepEqual(module, { id: 1, title: 'Module 1', type: 'module', descriptionFileName: 'module1/index.md' });
+			t.deepEqual(module, { id: 1, title: 'Module 1', type: 'module', descriptionFileName: 'module1/index.html' });
 			t.deepEqual(topic, manifest[1]);
 			t.is(data, '<html><head></head><body><a href="/d2l/le/lessons/123/units/1" target="_parent">Module 1</a></body></html>');
 		}
 	};
 
-	const rewriter = new LinkRewriter('/testdata', MockValenceApi, fs);
+	const fileHandler = new FileHandler('/testdata', mockFs);
+	const rewriter = new LinkRewriter(fileHandler, MockValenceApi);
+
+	await rewriter.rewriteLinks('https://example.com/', OrgUnit, manifest);
+});
+
+test.serial('rewriteLinks updates complex links', async t => {
+	mockFs({
+		'/testdata/module1/index1.html': '<html><body><a href="/module1/module2/topic4.md">Topic</a></body></html>',
+		'/testdata/module1/topic2.md': '<html><body><a href="./module2/topic4.md">Topic</a></body></html>',
+		'/testdata/module1/module2/index3.html': '<html><body><a href="../topic2.md">Topic</a></body></html>',
+		'/testdata/module1/module2/topic4.md': '<html><body><a href="../../module1/topic2.md">Topic</a></body></html>'
+	});
```

Comment:
Interesting test, I'm not sure we'd ever have a mix of HTML and markdown files though.

Reply:
It could happen  :)
I should probably change the .md file to have markdown though.  The main intent of the test is to exercise the relative links